src/test/java/com/snowflake/kafka/connector/internal/SinkServiceIT.java

```diff
@@ -111,11 +112,11 @@ public void testIngestion() throws Exception
       , "test", input.schema(), input.value(), offset);
 
     service.insert(record1);
-    List<String> files = conn.listStage(stage,
-      FileNameUtils.filePrefix(TestUtils.TEST_CONNECTOR_NAME, table,
-        partition));
-
-    assert files.size() == 1;
+    TestUtils.assertWithRetry(() -> conn.listStage(stage, FileNameUtils.filePrefix(TestUtils.TEST_CONNECTOR_NAME,
```

Comment:
Since insert does not synchronically trigger flush anymore, file will not appear on stage immediately after insert return. Therefore added the retry function here.